Expensify/App
# Use more user friendly errors when choosing default fund for Wallet

Related to https://github.com/Expensify/Web-Expensify/pull/33132

And related to small backend change where I emphasize the error messages need to be updated in frontend too if changed in backend https://github.com/Expensify/Auth/pull/6477

### Details


We have only been showing a generic error when setting a default fund for Expensify Wallet has failed. This change handles all the possible error and gives the user better feedback when it fails. Unfortunately, for most of these errors the users cannot do much on their own, but knowing the specific error message can help the support to identify the next steps to fix it.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7521

### Tests



*To test all the specific errors would be hard. We can leave this for Applause as they reported this problem and we are only making the error messages a bit more helpful.*

Prerequisite: Set the VBA in testing account (But make sure it is not in OPEN state)
1. Launch the app
2. Log in expensifail account (when locally testing any account which has Wallet)
3. Go to Setting - Payment
4. Tap on some card
5. Tap Set a default payment
6. Enter correct password
7. Tap Make Default payment method
8. Make sure there is more specific like this.

<img width="1173" alt="image" src="https://user-images.githubusercontent.com/36083550/155907197-8b607cea-58ac-437d-b52a-25cdd1bc6fbd.png">


### PR Review Checklist

#### Contributor (PR Author) Checklist

#### PR Reviewer Checklist

### QA Steps



Prerequisite: Set the VBA in testing account
1. Launch the app
2. Log in expensifail account
3. Go to Setting - Payment
4. Tap on Verified Account
5. Tap Set a default payment
6. Enter correct password
7. Tap Make Default payment method
8. One of these errors should show:
- 'There was an error setting this bank account as your default payment method.',
- 'This bank account is temporarily suspended.',
- 'There was an error setting this card as your default payment method.',
- 'Something went wrong. Please chat with Concierge for further assistance.',

### Tested On


### Screenshots


#### Web


<img width="1173" alt="image" src="https://user-images.githubusercontent.com/36083550/155907197-8b607cea-58ac-437d-b52a-25cdd1bc6fbd.png">

since this is only API error handling change, I have not tested it on all platforms as there is not expectation of change.
